lib/GenomeFileUtil/core/FastaGFFToGenome.py

```diff
@@ -793,31 +868,64 @@ def end(loc):
             ValueError('Feature {feature["id"]} must contain either exon or cds data to '
                        'construct an accurate location and sequence')
 
-    def _gen_genome_info(self, assembly_ref, assembly, input_gff_file, molecule_type, params):
+    '''
+    Metagenome Changes:
+        Here is the meat of the saving operation.
+    '''
+    def _gen_genome_info(self, assembly_ref, assembly, input_gff_file, molecule_type, prot_fasta_path, params):
         """
         _gen_genome_info: generate genome info
 
+        Genome Fields:
+            features: protein encoding genes
+            cdss:
+            mrnas: mrna sequences
+            non_coding_features: everything that doesn't fall into 'features',
+                'cdss', 'mrnas'
         """
+        features = []
+        cdss = []
+        mrnas = []
+        non_coding_features = []
         genome = {
             "id": params.get('genome_name'),
             "scientific_name": params.get('scientific_name', "Unknown"),
             "assembly_ref": assembly_ref,
             'molecule_type': molecule_type,
-            "features": [],
-            "cdss": [],
-            "mrnas": [],
-            'non_coding_features': [],
             "gc_content": assembly["gc_content"],
             "dna_size": assembly["dna_size"],
             'md5': assembly['md5'],
             'num_contigs': len(assembly['contigs']),
             'ontologies_present': dict(self.ontologies_present),
             'ontology_events': self.ontology_events,
         }
+        if self.is_metagenome:
+            metagenome_fields = [
+                ("publications", []),
+                ("external_source_origination_date", None),
+                ("original_source_file_name", None),
+                ("notes", None),
+                ("environment" , None),
```

Comment:
I'm not sure, but those fields were included in the Project Description for the datatype. I left them as input parameters because I wasn't aware of how else I would get them.

Reply:
Seems worth some thought especially with the "environment" variable which could easily follow some sort of fixed lexicon